tests/analysis/test_analysis.py

```diff
@@ -251,3 +256,49 @@ def test_get_inference_results(experiment_analysis, analysis_df_diff):
     assert results.ate > 0
     assert 0 <= results.p_value < 1
     assert results.std_error > 0
+
+
+def test_delta_analysis(analysis_ratio_df, experiment_dates):
+    analyser = DeltaMethodAnalysis(cluster_cols=["user"], scale_col="scale")
+    experiment_start = min(experiment_dates)
+
+    df_experiment = analysis_ratio_df.query(f"date >= '{experiment_start}'")
+
+    assert 0.05 >= analyser.get_pvalue(df_experiment) >= 0
+
+
+def test_aa_delta_analysis(dates):
+
+    analyser = DeltaMethodAnalysis(cluster_cols=["user"], scale_col="scale")
+
+    p_values = []
+    for _ in range(1000):
+        data = generate_ratio_metric_data(dates, N=100_000, treatment_effect=0)
+        p_values.append(analyser.get_pvalue(data))
+
+    positive_rate = sum(p < 0.05 for p in p_values) / len(p_values)
+
+    assert positive_rate == pytest.approx(
+        0.05, abs=0.01
+    ), "P-value A/A calculation is incorrect"
+
+
+def test_stats_delta_vs_ols(analysis_ratio_df, experiment_dates):
```

Comment:
I'd also test that you get the same results with aggregated and non-aggregated data

Reply:
yep, be4-hand

Reply:
maybe it's not needed, I just think we need to test the aggregation in some way (I think we're not testing it), wdyt is best?